src/SmartTransactionsController.ts

```diff
@@ -840,9 +894,14 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
   #getChainId({
     networkClientId,
   }: { networkClientId?: NetworkClientId } = {}): Hex {
-    return networkClientId
-      ? this.getNetworkClientById(networkClientId).configuration.chainId
-      : this.config.chainId;
+    if (networkClientId) {
+      return this.messagingSystem.call(
+        'NetworkController:getNetworkClientById',
+        networkClientId,
+      ).configuration.chainId;
+    }
```

Comment:
Do you think this is still necessary since we change `this.chainId` everytime `NetworkController` emits a state update?

Reply:
I believe we'll still need that logic since `#getChainId` is intended to return the corresponding `chainId` for any `networkClientId`, not just the `selectedNetworkClientId`.